src/notesubgroup.ts

```diff
@@ -55,45 +63,48 @@ export class NoteSubGroup extends Modifier {
     }).setStrict(false);
 
     this.voice.addTickables(this.subNotes);
-
-    return this;
   }
 
-  getCategory() {
+  getCategory(): string {
     return NoteSubGroup.CATEGORY;
   }
 
-  preFormat() {
+  preFormat(): void {
     if (this.preFormatted) return;
 
     this.formatter.joinVoices([this.voice]).format([this.voice], 0);
     this.setWidth(this.formatter.getMinTotalWidth());
     this.preFormatted = true;
   }
 
-  setNote(note) {
+  setNote(note: Note): this {
     this.note = note;
+    return this;
   }
-  setWidth(width) {
+
+  setWidth(width: number): this {
     this.width = width;
+    return this;
   }
-  getWidth() {
+
+  getWidth(): number {
     return this.width;
   }
 
-  draw() {
+  draw(): void {
     this.checkContext();
```

Comment:
Get the `ctx` here, and you can delete the new line 107 below:
`const ctx = this.checkContext()`

Reply:
thanks! wasn't aware that checkContext() returned the context, haha.